### ISSUE TITLE
Ember Data decorators do not appear to work with ES6 classes

### ISSUE THREAD

**Issue by ebryn** [rwjblue/ember-computed-decorators#98]
```
import DS from 'ember-data';
import { attr } from 'ember-computed-decorators/ember-data';

export default class extends DS.Model {
  @attr title;
};
```

Using Babel 6 with this EmberApp config:
```
    babel: {
      plugins: [
        'transform-decorators-legacy',
        "transform-class-properties"
      ]
    }
```

**Comment by pzuraq** [comment]
I think this is blocked by some pending changes in Ember Data

cc @runspired

**Comment by NullVoxPopuli** [comment]
decorators currently don't work at all with ember-cli-babel 6. 
see:
https://github.com/rwjblue/ember-computed-decorators/issues/96
https://github.com/rwjblue/ember-computed-decorators/issues/93
https://github.com/rwjblue/ember-computed-decorators/issues/94
https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy/issues/65
and kinda: https://github.com/babel/ember-cli-babel/issues/69

People seem to be content with adding `: null` to all their decorators. :-1: 

I've tried `babel-plugin-transform-decorators` -- also doesn't work.

I have yet to find a way to get decorators working without adding 130 `: null`'s

**Comment by pzuraq** [comment]
The null fix is for the standard Ember object model, when you are using `extend`. When used in ES6 classes null isn't needed, and it works generally, but not for DS.Model yet.

**Comment by NullVoxPopuli** [comment]
does this only work with components?
I tried:
```js
export default class CurrentUserService extends Ember.Service { ...
``
and I get a Broccoli error

**Comment by rwjblue** [comment]
Not sure why you got a syntax error, but it does work generally speaking...

**Comment by NullVoxPopuli** [comment]
@rwjblue do you have an opensource project that I can take a gander at dependencies and configs and see if I can find what I'm missing?

**Comment by NullVoxPopuli** [comment]
found out I needed babel-plugin-transform-class-properties. everything is working okish, and everything is more modern now :-)

**Comment by runspired** [comment]
I don't have to use : null to use decorators.. likely a config issue here

**Comment by pzuraq** [comment]
With babel 6 and the `transform-decorators-legacy` plugin, properties must have a value when using decorators on an object:

```js
export default Ember.Object.extend({
  @alias('foo') bar, // invalid
  @alias('foo') baz: null, // valid

  @computed
  foo() {
    // valid, because the function is the value
  }
});
```

This isn't an issue with ES6 classes and class properties, and hopefully won't be when the actual `transform-decorators` plugin is finished

**Comment by NullVoxPopuli** [comment]
hm. I'm running in to a thing, where my ES6 decorated properties don't show up at all.

![2017-06-26-111454_515x381_scrot](https://user-images.githubusercontent.com/199018/27546418-b3d88720-5a60-11e7-8980-212cbc3fad49.png)

and then in chrome inspector:

![2017-06-26-111645_528x841_scrot](https://user-images.githubusercontent.com/199018/27546512-fb0d123c-5a60-11e7-8c28-958a84707f6b.png)

**Comment by pzuraq** [comment]
Which version of Ember and computed decorators are you using? This should be fixed on master if it is what I think it is

**Comment by NullVoxPopuli** [comment]
from package.json:
ember-computed-decorators: 0.3.0,
ember-source: 2.13.0

**Comment by pzuraq** [comment]
Hmm, interesting. There was a bug where the actual constructor of the class would never actually run due to the double extend uses for injection of previous versions of Ember, but that shouldn't be an issue on 2.13.

Can you try it against master and if it's still an problem open up another issue for it?

**Comment by NullVoxPopuli** [comment]
setting my version of ember-computed-decorators to `rwjblue/ember-computed-decorators` fixed it! woo! 

thanks, @pzuraq !

**Comment by NullVoxPopuli** [comment]
I've upgraded to ember-decorators, and am trying using an es6 class:

```js
import Ember from 'ember';
import DS from 'ember-data';

import { computed, readOnly } from 'ember-decorators/object';
import { hasMany, attr } from 'ember-decorators/data';

import { userIsMemberOf } from 'aeonvera/helpers/user/is-member-of';
import { userLatestRenewalFor } from 'aeonvera/helpers/user/latest-renewal-for';

export default class extends DS.Model {
  @hasMany('membership-renewal') membershipRenewals
}
```

I get the error (with and without the specified model name in hasMany): 
```
TypeError: Cannot read property 'modelName' of undefined
    at Function._findInverseFor (model.js:662)
    at Function.inverseFor (model.js:629)
    at createRelationshipFor (create.js:45)
    at Relationships.get (create.js:88)
    at store.js:299
    at model.js:882
    at cb (ember.debug.js:21393)
    at OrderedSet.forEach (ember.debug.js:21194)
    at Map.forEach (ember.debug.js:21397)
    at Function.eachRelationship (model.js:881)
```


Using the non es6 class / `extend` method works.

**Comment by pzuraq** [comment]
Yeah, this issue is dependent on work in Ember Data. Hopefully it'll come through in the next few releases.

**Comment by rwjblue** [comment]
I think we might be able to work around the issue, but a failing test PR would be super helpful.

**Comment by NullVoxPopuli** [comment]
@rwjblue I 'think' I made a failing test -- messed up my local env, though. Looks like this repo could use some continuous testing.

**Comment by pzuraq** [comment]
v2.0.0-beta.1 has been released and should fix these issues! Let me know if you have any problems, I'll close this issue once we get to v2.0.0 stable
